docs/element-definitions.md

```diff
@@ -346,16 +381,32 @@ A reference:
 
 ---
 
+### Fail Element
```

Comment:
I've never see a fail element and I do not understand the point of it if instantiation of it would fail it cannot be used. Could you please elaborate?

Reply:
It can be inserted in places where otherwise there would be an error. For example, an object referencing itself in a member resembles a circular reference; instead of just failing, we could insert a Fail Element and allow tooling to recover from the issue.

Reply:
I introduced this to reserve a keyword for such a type; based on discussion with Pavan. But we can leave it out.

Reply:
We decided not to actually serialize it. Just say "Type with empty domain" and say it's reserved